src/config/config.ts

```diff
@@ -614,27 +622,35 @@ class SfdxConfig {
     return Object.assign(config, sfdxConfig);
   }
 
-  public readSync(): ConfigProperties {
+  public async write(config = this.config.toObject()) {
     try {
-      const contents = parseJsonMap(fs.readFileSync(this.getSfdxPath(), 'utf8'));
-      return this.normalize(contents as ConfigProperties, 'toNew');
+      const translated = this.translate(config as ConfigProperties, 'toOld');
+      await this._write(translated);
     } catch (error) {
       /* Do nothing */
-      return {};
     }
   }
 
-  public async writeSync(config = this.config.toObject()) {
+  private readSync(): ConfigProperties {
```

Comment:
why is there `_readSync` and `readSync`? Same for write?

Reply:
I originally did it for some unit tests that I had written for my initial solution. However, those tests are no longer needed since I scraped that solution. But yeah, I can revert this change